jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -108,18 +141,42 @@ public static boolean getDefaultUseCaches()
      */
     public static Resource.Mount mountIfNeeded(URI uri)
     {
-        if (uri == null || uri.getScheme() == null)
+        if (uri == null)
+            return null;
+        String scheme = uri.getScheme();
+        if (scheme == null)
+            return null;
+        if (!isArchive(uri))
             return null;
         try
         {
-            return (uri.getScheme().equalsIgnoreCase("jar")) ? FileSystemPool.INSTANCE.mount(uri) : null;
+            if (scheme.equalsIgnoreCase("jar") || scheme.equalsIgnoreCase("file"))
+            {
+                URI jarURI = toJarFileUri(uri);
+                if (jarURI == null)
+                    throw new IllegalArgumentException("Not a jar or file URI: " + uri);
+                return FileSystemPool.INSTANCE.mount(jarURI);
+            }
+            return null;
         }
         catch (IOException ioe)
         {
             throw new IllegalArgumentException(ioe);
         }
     }
 
+    /**
+     * @param resourceBase the base resource that might need to be mounted.
+     * @return the {@link Mount} if a mount is needed, null if one is not needed.
+     * @see #mountIfNeeded(URI)
+     */
+    public static Mount mountIfNeeded(Resource resourceBase)
```

Comment:
That looks weird: you should not be able to create a `Resource` that needs mounting.

Reply:
Method removed